integration/db/db_integration_test.go

```diff
@@ -563,6 +563,13 @@ func TestDatabaseRootLeafIdleTimeout(t *testing.T) {
 
 	t.Run("root role with idle timeout", func(t *testing.T) {
 		setRoleIdleTimeout(t, rootAuthServer, rootRole, idleTimeout)
+		require.Eventually(t, func() bool {
+			role, err := rootAuthServer.GetRole(context.Background(), rootRole.GetName())
+			require.NoError(t, err)
```

Comment:
The condition passed to `Eventually` runs in a background goroutine, so you can't `require.NoError` in there.

Reply:
Right, Thanks will fix.

Reply:
@zmb3 why does something running in a go routine mean that `require.NoError` can't be called from it? Glancing through the codebase, it looks like we're doing that quite often elsewhere.

Reply:
`require` calls `FailNow`.

```
➜ go doc testing.T.FailNow
package testing // import "testing"

func (c *T) FailNow()
    FailNow marks the function as having failed and stops its execution by
    calling runtime.Goexit (which then runs all deferred calls in the current
    goroutine). Execution will continue at the next test or benchmark. FailNow
    must be called from the goroutine running the test or benchmark function,
    not from other goroutines created during the test. Calling FailNow does not
    stop those other goroutines.
```

Yes, we do this wrong in lots of places. It's an easy mistake to make. Easiest fix is to use `assert.NoError` instead.